### PR TITLE
Product card

### PR DESCRIPTION
**Pull request by navid9696**
Closes: #15

### CODE REVIEW

File: src/components/AccountSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	const [modalOpen, setModalOpen] = useState(false)
+
+	const handleModalClose = useCallback(() => {
+		setModalOpen(false)
+	}, [])
+
+	const handleOpenSettings = useCallback((id: string) => {
+		setContentId(id)
+		setIsFormVisible(true)
+	}, [])
```
Comment (bart-krakowski) [inline src/components/AccountSettings.tsx:38]: `useCallbacks` are redundant here

File: src/components/EmailForm.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import { useState, useEffect } from 'react'
+import { toast } from 'react-toastify'
+import { SettingFormsProps } from '~/lib/types'
+
+const emailSchema = z.object({
+	email: z
+		.string()
+		.min(1, { message: 'Your email address cannot be empty.' })
+		.email({ message: 'Please enter a valid email address.' }),
+})
```
Comment (bart-krakowski) [inline src/components/EmailForm.tsx:16]: Schemas

File: src/components/ExchangeInput.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { ChangeEventHandler } from 'react'
+import { FormControl, InputAdornment, InputLabel, OutlinedInput } from '@mui/material'
+
+interface ExchangeInputProps {
+	icon: React.ReactNode
+	inputProps?: { [key: string]: string | number | boolean | null }
```
Comment (bart-krakowski) [inline src/components/ExchangeInput.tsx:6]: This definition is not type-safe. Pls use the proper interface (`InputBaseComponentProps`)

File: src/components/FormBase.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+const FormBase = ({ title, subtitle, size, className, fields, resolver, onSubmitSuccess, children }: FormBaseProps) => {
+	const [focusedField, setFocusedField] = useState<string | null>(null)
+	const {
+		register,
+		handleSubmit,
+		formState: { errors },
+		clearErrors,
+	} = useForm<any>({
```
Comment (bart-krakowski) [inline src/components/FormBase.tsx:29]: `any`

File: src/components/NicknameForm.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+import { SettingFormsProps } from '~/lib/types'
+
+const nicknameSchema = z.object({
+	nickname: z
+		.string()
+		.min(1, { message: 'Your nickname cannot be empty.' })
+		.regex(/^[a-zA-Z]/, { message: 'Start your name with a letter.' })
+		.min(5, { message: 'Make sure your name is at least 5 characters long.' })
+		.max(15, { message: 'Keep your name within 15 characters.' }),
+})
```
Comment (bart-krakowski) [inline src/components/NicknameForm.tsx:18]: Schemas

File: src/components/NicknameForm.tsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+		console.log(data)
+	}
+
+	useEffect(() => {
+		if (formState.isSubmitSuccessful) {
+			setIsfocusedField(false)
+			toast.success('Nickname updated successfully')
+			reset()
+		}
+	}, [formState, reset])
```
Comment (bart-krakowski) [inline src/components/NicknameForm.tsx:47]: Same thing with calling toast in the `useEffect` hook

File: src/components/PasswordForm.tsx [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+const updatePasswordSchema = z
+	.object({
+		currentPassword: z.string().min(1, { message: 'Current password is required' }),
+		password: passwordSchema,
+		confirmPassword: z.string(),
+	})
+	.refine(data => data.password === data.confirmPassword, {
+		message: "The passwords you entered don't match. Please verify and try again.",
+		path: ['confirmPassword'],
+	})
```
Comment (bart-krakowski) [inline src/components/PasswordForm.tsx:37]: Schemas

**Review by bart-krakowski — CHANGES_REQUESTED** [review]

File: src/components/NicknameForm.tsx (hunk 6 above)
Comment (navid9696) [reply src/components/NicknameForm.tsx:47]: useEffect is used temporarily, when I will start creating database integration I intend to change it of course, probably with tRPC

File: src/components/AccountSettings.tsx (hunk 1 above)
Comment (navid9696) [reply src/components/AccountSettings.tsx]: resolved in account-settings branch

File: src/components/EmailForm.tsx (hunk 2 above)
Comment (navid9696) [reply src/components/EmailForm.tsx]: resolved in account-settings branch

File: src/components/FormBase.tsx (hunk 4 above)
Comment (navid9696) [reply src/components/FormBase.tsx]: resolved in account-settings branch

File: src/components/NicknameForm.tsx (hunk 5 above)
Comment (navid9696) [reply src/components/NicknameForm.tsx]: resolved in account-settings branch

File: src/components/PasswordForm.tsx (hunk 7 above)
Comment (navid9696) [reply src/components/PasswordForm.tsx]: resolved in account-settings branch

**Pull request merged**